docs/source/glm.rst

```diff
@@ -180,11 +184,8 @@ available link functions can be obtained by
    NegativeBinomial
    Power
    cauchy
-   cloglog
    identity
    inverse_power
    inverse_squared
-   log
-   logit
```

Comment:
part of this was ancient compatibility and refactoring

log and logit are subclasses of the capitalized that don't change anything, i.e. identical and redundant classes
some other lower case like inverse_power are subclasses that fix a parameter (e.g. power)

There is a partial refactoring in master but not in 0.8, for how links are handled, but no name deprecation yet.
AFAIR, most of our examples use the lowercase `log` but in future we might want to switch to the uppercase class names.